test/local_test.go

```diff
@@ -0,0 +1,74 @@
+package test
+
+import (
+	"errors"
+	"gopkg.in/yaml.v3"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type parsedCRD struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Annotations struct {
+			EksAmazonawsComRoleArn              string `yaml:"eks.amazonaws.com/role-arn"`
+			EksAmazonawsComStsRegionalEndpoints string `yaml:"eks.amazonaws.com/sts-regional-endpoints"`
+		} `yaml:"annotations"`
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestServiceAccountsUseStsRegionalEndpoint(t *testing.T) {
+	addonsPath := "../modules/cluster/addons/"
+	addonFiles, err := os.ReadDir(addonsPath)
+	if err != nil {
+		t.Error(err)
+	}
+	for _, v := range addonFiles {
+		fileName := v.Name()
+		if filepath.Ext(fileName) == ".yaml" {
+			crds, err := readManifest(addonsPath + fileName)
+			if err != nil {
+			    // do something for error handling
+			    // e.g
+			    // t.Fatalf("err=%#v\n" , err)
+			}
+			for _, v := range crds {
+				if v.Kind == "ServiceAccount" {
```

Comment:
I don't think you need to double bag this

https://github.com/cookpad/terraform-aws-eks/pull/249/files#diff-b1a079e8d2b72d05971b0c6d094ba1f8adc1ee7528e4f3ae70da8c4aec63294fR68

Reply:
By that, I mean, you populated crds with only things that had a Kind of ServiceAccount, I don't think you need to filter here.